backend/src/v2/component/launcher_v2.go

```diff
@@ -472,47 +522,49 @@ func uploadOutputArtifacts(ctx context.Context, executorInput *pipelinespec.Exec
 		if len(artifactList.Artifacts) == 0 {
 			continue
 		}
-		// TODO: Support multiple artifacts someday, probably through the v2 engine.
-		outputArtifact := artifactList.Artifacts[0]
 
-		// Merge executor output artifact info with executor input
-		if list, ok := executorOutput.Artifacts[name]; ok && len(list.Artifacts) > 0 {
-			mergeRuntimeArtifacts(list.Artifacts[0], outputArtifact)
-		}
+		for _, outputArtifact := range artifactList.Artifacts {
```

Comment:
This whole block only looks like a change because it's intended. We're just actually iterating through the artifacts now instead of taking the first one from the list.